### DISCUSSION TITLE
[V5] Return to the page before login (dynamic redirect)?

### DISCUSSION THREAD

**Discussion by roliveira3891** [adonisjs/core#1766] [Help]
Hello everyone. How can I return to the page I was on before login (dynamic redirect)?

another thing: when the user is not logged in, how to redirect to the login page automatically?

Thank you

**Comment by DraftProducts** [comment]
Hi @roliveira3891, most of the time, the best way is to add a redirect query in the url when the Exeption Handler redirect your user to your login page (example: `?redirect=/home`).
So when the login is success you can redirect the user to the query (the desired url).

**Reply by roliveira3891** [reply]
Sorry for the delay.

Configuring the Middleware Auth? 

Example; Route().Middleware('auth) => redirect /login

Thank you
